stdio/format.c

```diff
@@ -483,7 +479,21 @@ static void format_sprintf_num(void *ctx, feedfunc feed, uint64_t num64, uint32_
 		}
 	}
 
-	const int digits_cnt = tmp - tmp_buf;
+	if (((flags & FLAG_OCT) != 0) && ((num64 != 0) || (float_frac_len == 0)) && ((flags & FLAG_ALTERNATE) != 0)) {
+		*tmp++ = '0';
+	}
+
+	if (tmp - tmp_buf < float_frac_len) {
+		memset(tmp, '0', float_frac_len - (tmp - tmp_buf));
+		tmp = tmp_buf + float_frac_len;
+	}
+
+	if (((flags & FLAG_HEX) != 0) && (num64 != 0) && ((flags & FLAG_ALTERNATE) != 0)) {
+		memcpy(tmp, prefix, 2);
+		tmp += 2;
```

Comment:
```suggestion
		tmp[0] = prefix[0];
		tmp[1] = prefix[1];
		tmp += 2;
```
`tmp` and `prefix` are `char` type, it is faster.